src/spec-common/injectHeadless.ts

```diff
@@ -105,13 +105,18 @@ export type DevContainerConfigCommand = 'initializeCommand' | 'onCreateCommand'
 
 const defaultWaitFor: DevContainerConfigCommand = 'updateContentCommand';
 
+export interface DevContainerFeature {
+	id: string;
+	options: boolean | string | Record<string, boolean | string | undefined>;
+}
+
 export interface CommonDevContainerConfig {
 	configFilePath?: URI;
 	remoteEnv?: Record<string, string | null>;
 	forwardPorts?: (number | string)[];
 	portsAttributes?: Record<string, PortAttributes>;
 	otherPortsAttributes?: PortAttributes;
-	features?: Record<string, string | boolean | Record<string, string | boolean>>;
+	features?: DevContainerFeature[] | Record<string, string | boolean | Record<string, string | boolean>>;
```

Comment:
We don't want to support the array variant. Please see https://github.com/devcontainers/spec/pull/27/files#r869018039.

Reply:
Made a comment on that thread.

Reply:
I have shared some thoughts as well

Reply:
Dropping my thoughts here since the spec PR has already been merged for a couple weeks. I strongly believe we need to support the array syntax, and I would go a step further and say that we should only document the array syntax going forward, leaving the object syntax in place only for backwards compatibility (object can be easily mapped to the array form at runtime in a deterministic way).

There are several scenarios we've encountered from users who need to better control feature execution order as a consumer of the feature in a way that the feature author wouldn't know about in advance. Most of these involve custom logic that the consumer wants to run after a feature is installed, but are things the user wants baked into the image itself and not later on in a lifecycle hook.  A few of the examples:
* Running a tool from an earlier feature to install other tools in a later feature
    * A user wants to install a global npm package (`npm install -g`) after Node/npm are installed
    * A user wants to install some Python packages through pip after installing python/pip
    * A user wants to install an Azure CLI extension after installing the Azure CLI
    * A user wants to install a GH CLI extension after installing the GH CLI

I understand the argument that the tooling should be able to figure out the right order if dependencies are defined, but I don't think we have a solution for dependency declaration within each feature that is robust enough to handle all of the cases. Ultimately, giving users the ability to force a specific ordering puts them in control and allows them to make adjustments as needed instead of being at the mercy of the tooling to figure out the perfect install order.

For most users, they won't need to worry about the installation order of different features, and in that case, both syntaxes work just fine. But in the case where users do need the control for the ordering, the array syntax makes that possible, whereas the object syntax does not.